Jenkinsfile.macos

```diff
@@ -1,175 +1,282 @@
+def utils
+
 pipeline {
 
   agent { label 'macos-v1.4-arm64' }
 
   options {
+    // timeout regardless of activity
+    timeout(activity: false, time: 120)
     timestamps()
     disableConcurrentBuilds()
     buildDiscarder(logRotator(numToKeepStr: '100'))
   }
 
   parameters {
-      string(name: 'RSTUDIO_VERSION_MAJOR',  defaultValue: '1', description: 'RStudio Major Version')
-      string(name: 'RSTUDIO_VERSION_MINOR',  defaultValue: '1', description: 'RStudio Minor Version')
-      string(name: 'RSTUDIO_VERSION_PATCH',  defaultValue: '999', description: 'RStudio Patch Version')
-      string(name: 'RSTUDIO_VERSION_SUFFIX', defaultValue: '+9999', description: 'RStudio Pro Suffix Version')
-      string(name: 'GIT_REVISION', defaultValue: 'main', description: 'Git revision to build')
-      string(name: 'BRANCH_NAME', defaultValue: 'origin/main', description: 'Branch name to build')
       string(name: 'SLACK_CHANNEL', defaultValue: '#ide-builds', description: 'Slack channel to publish build message.')
+      string(name: 'RSTUDIO_VERSION_PATCH',  defaultValue: '999', description: 'RStudio Patch Version')
+      booleanParam(name: 'DAILY', defaultValue: false, description: 'Runs daily build if true')
+      booleanParam(name: 'PUBLISH', defaultValue: true, description: 'Runs publish stage if true')
+      gitParameter defaultValue: "${env.GIT_BRANCH}",
```

Comment:
@MariaSemple Is this what we should be doing (along with the "Checkout" stage below) for builds that are downstream of the job dispatcher? When I ran this this pipeline in my tests standalone, the `COMMIT_HASH` parameter ended up being null, so I had to comment it out to get it working, but I figured that this should work when the param is passed from an upstream job.

Reply:
Yep, that looks right to me!